### ISSUE TITLE
Support backslash escapes

### ISSUE THREAD

**Issue by inexorabletash** [inexorabletash/uate#1]
There's no simple way to write a literal `${` in a uate string. In ES6 you'd write:

``` js
`This is not a substitution: \${example}.`
```

uate should support backslashes, which would end up looking like this:

``` js
eval(uate('This is not a substitution: \\${example}.')))
```

(Double backslash is needed otherwise it's eaten by the JS parser.)

**Comment by inexorabletash** [comment]
Fixed in 4ace17d6f55162c00807e58fe872b2737388abb5 but need tests.

**Comment by inexorabletash** [comment]
tests added in e0be96aed55e9314c7769dc6a398756d011a94b9
